components/board/collector.go

```diff
@@ -26,51 +27,28 @@ func (m method) String() string {
 	return "Unknown"
 }
 
-// AnalogRecords a collection of AnalogRecord.
-type AnalogRecords struct {
-	Readings []AnalogRecord
-}
-
-// AnalogRecord a single analog reading.
-type AnalogRecord struct {
-	AnalogName  string
-	AnalogValue int
-}
-
-// GpioRecords a collection of GpioRecord.
-type GpioRecords struct {
-	Readings []GpioRecord
-}
-
-// GpioRecord a signle gpio reading.
-type GpioRecord struct {
-	GPIOName  string
-	GPIOValue bool
-}
-
 func newAnalogCollector(resource interface{}, params data.CollectorParams) (data.Collector, error) {
 	board, err := assertBoard(resource)
 	if err != nil {
 		return nil, err
 	}
 
 	cFunc := data.CaptureFunc(func(ctx context.Context, arg map[string]*anypb.Any) (interface{}, error) {
-		var readings []AnalogRecord
-		for k := range arg {
-			if reader, ok := board.AnalogReaderByName(k); ok {
-				value, err := reader.Read(ctx, data.FromDMExtraMap)
-				if err != nil {
-					// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
-					// is used in the datamanager to exclude readings from being captured and stored.
-					if errors.Is(err, data.ErrNoCaptureToStore) {
-						return nil, err
-					}
-					return nil, data.FailedToReadErr(params.ComponentName, analogs.String(), err)
+		var value int
+		if reader, ok := board.AnalogReaderByName(arg["reader_name"].String()); ok {
```

Comment:
What was the outcome here? https://github.com/viamrobotics/rdk/pull/2989#discussion_r1341567521

Reply:
Still need to check, will do today.

Reply:
It does handle it gracefully in that there is no error thrown. It will just return nil. Should I add a case where if it is nil, itll error as the config would be invalid?

Reply:
Just to clarify - if it is nil, that means the config doesn't actually contain an AnalogReader with the passed `reader_name`, so we can't capture anything? If so, I think yeah returning a FailedToReadErr indicating the reader doesn't exist would make sense

Reply:
Yes that is correct. Ill update that.